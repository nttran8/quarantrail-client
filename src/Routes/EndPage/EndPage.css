.middle1 {
    position: fixed; 
    width: 80%;
    height:40%;
    top: 150px;
    margin-left: 10%;
    margin-right:10%;
    display: flex;
    align-items: middle;
    justify-content: center;
    flex:1 1;
  }
  .box2 {
    display: flex;
    flex:1 1;
    align-items: center;
    border:1px solid black;
    width: 75%;
    height: 100%;
    flex-direction: column;
    justify-content: space-evenly;
  
}