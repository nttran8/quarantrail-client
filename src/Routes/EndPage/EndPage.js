import React, { Component } from "react";
import { Link, Redirect } from "react-router-dom";
import PersonContext from "../../Context/PersonContext";
import LeaderboardService from "../../services/leaderboard-service";
import Music from "../../Components/Music/Music";
import Song from "../../Sound/seeyouagain.mp3";
import "./EndPage.css";
export default class EndPage extends Component {
  static contextType = PersonContext;

  handleRestart = () => {
    this.context.resetDay();
    const post = {
      name: this.context.name,
      score: this.context.day
    };
    this.context.updateRenderCurve(false);
    LeaderboardService.postScore(post);
    this.props.renderRestart();
    return <Redirect to="/" />;
  };
  render() {
    let day;
    if (this.context.day === 1) {
      day = "day";
    } else day = "days";
    return (
      <section className="EndPage-section">
        <div className="display-container">
          <div className="display">
            <h1>{this.props.location.state.note}</h1>
            <h2>
              Score: {this.context.day} {day}
            </h2>
            <Link to="/">
              <button className="display-button" onClick={this.handleRestart}>
                Play again
              </button>
            </Link>
            <Music song={Song} />
          </div>
        </div>
        <Music song={Song} />
      </section>
    );
  }
}
