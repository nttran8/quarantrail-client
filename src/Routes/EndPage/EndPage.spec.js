import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EndPage from './EndPage'

describe(`EndPage Route`, () => {

  it('renders by default', () => {
    const wrapper = shallow(<EndPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})