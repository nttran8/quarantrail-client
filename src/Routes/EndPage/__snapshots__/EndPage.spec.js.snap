// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndPage Route renders by default 1`] = `
<section
  className="EndPage"
>
  <div
    class="middle1"
  >
    <div
      class="box2"
    >
      <h1 />
      <h2>
        you have survived: 
         days
      </h2>
      <Link
        to="/"
      >
        <button
          onClick={[Function]}
        >
          Try again
        </button>
      </Link>
      <Music
        song="seeyouagain.mp3"
      />
    </div>
  </div>
</section>
`;
